pkg/controllers/node/termination/terminator/eviction.go

```diff
@@ -192,10 +199,35 @@ func (q *Queue) Evict(ctx context.Context, key QueueKey) bool {
 		log.FromContext(ctx).Error(err, "failed evicting pod")
 		return false
 	}
-	q.recorder.Publish(terminatorevents.EvictPod(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: key.Name, Namespace: key.Namespace}}))
+	q.recorder.Publish(terminatorevents.EvictPod(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: key.Name, Namespace: key.Namespace}}, evictionMessage))
 	return true
 }
 
+func getNodeClaims(ctx context.Context, key QueueKey, kubeClient client.Client) ([]*v1.NodeClaim, error) {
+	nodeClaimList := &v1.NodeClaimList{}
+	if err := kubeClient.List(ctx, nodeClaimList, client.MatchingFields{"status.nodeName": key.nodeName}); err != nil {
+		return nil, fmt.Errorf("listing nodeClaims, %w", err)
+	}
+	return lo.ToSlicePtr(nodeClaimList.Items), nil
+}
+
+func evictionReason(ctx context.Context, key QueueKey, kubeClient client.Client) (string, error) {
+	nodeClaims, err := getNodeClaims(ctx, key, kubeClient)
+	if err != nil {
+		return "", err
+	}
+	for _, nodeClaim := range nodeClaims {
+		terminationCondition := nodeClaim.StatusConditions().Get(v1.ConditionTypeDisruptionCandidate)
+		if terminationCondition.IsTrue() {
+			return terminationCondition.Message, nil
+		}
+		if !nodeClaim.DeletionTimestamp.IsZero() {
+			return fmt.Sprintf("node %s/%s is marked for deletion", nodeClaim.Name, nodeClaim.Status.NodeName), nil
```

Comment:
probably would want to pick one or the other here, including just the nodeclaim name or node name
```suggestion
			return fmt.Sprintf("nodeclaim %s/%s is marked for deletion", nodeClaim.Name, nodeClaim.Status.NodeName), nil
```